hw/ip/spi_device/doc/_index.md

```diff
@@ -896,6 +896,41 @@ Due to the CDC issue, the SW is only permitted to update the registers when the
 1. When the FIFO is empty, the TPM sends `START` to the host system, receives the payload, and stores the data into the write FIFO.
 1. The SW, in the meantime, reads TPM_CMD_ADDR then reads the write FIFO data when the FIFO is available.
 
+### TPM_CMDADDR_NOTEMPTY Interrupt
+
+`TPM_CMDADDR_NOTEMPTY` interrupt remains high even SW clears the interrupt unless the cause is disappeared.
+SW should mask the interrupt if SW wants to process the event in a deferred way.
+
+```c
+isr_return_t spi_tpm_isr() {
+  uint32_t irq_deferred = 0;
+  uint32_t irq_status = spi_tpm_get_irq_status();
+  if (irq_status & kSpiTpmFifoIrq) {
+    irq_deferred |= kSpiTpmFifoIrq;
+    schedule_deferred_work(spi_tpm_deferred_work);
+  }
+  // ...
+  spi_tpm_mask_irq(irq_deferred);
+  return IRQ_HANDLED;
```

Comment:
nit: Since this is going into our repo, maybe remove the return value here (and make the function `void`). `IRQ_HANDLED` might be a defined constant in a real RTOS, and we don't mean to imply any specific meaning here. I don't know why I added it into the comment :smile: